modules/tinymce/src/core/test/ts/browser/DisabledModeTest.ts

```diff
@@ -355,12 +355,40 @@ describe('browser.tinymce.core.DisabledModeTest', () => {
     it('TINY-11488: Should not allow switching modes when the editor is disabled', async () => {
       const editor = hook.editor();
       await Waiter.pTryUntil('Wait for editor to be disabled', () => assertEditorDisabled(editor));
+
+      editor.mode.set('readonly');
+      await Waiter.pTryUntil('Wait for editor to be disabled', () => assertEditorDisabled(editor));
+
+      editor.options.set('disabled', false);
+      await Waiter.pTryUntil('Wait for editor to be enabled', () => assertEditorEnabled(editor));
+
+      editor.mode.set('readonly');
+      await Waiter.pTryUntil('Wait for editor to be enabled', () => assertEditorEnabled(editor));
```

Comment:
Should it assert mode to be readonly?

Reply:
It's checked below, but I've removed the duplicated test case 👍